payjoin/tests/integration.rs

```diff
@@ -200,20 +200,24 @@ mod integration {
         let receiver_substitute_address =
             receiver.get_new_address(None, None).unwrap().assume_checked();
         payjoin.substitute_output_address(receiver_substitute_address);
-
-        let payjoin_proposal_psbt = payjoin.apply_fee(None).unwrap();
-
-        // Sign payjoin psbt
-        let payjoin_base64_string = base64::encode(&payjoin_proposal_psbt.serialize());
-        let payjoin_proposal_psbt = receiver
-            .wallet_process_psbt(&payjoin_base64_string, None, None, Some(false))
-            .unwrap()
-            .psbt;
-        let payjoin_proposal_psbt = Psbt::from_str(&payjoin_proposal_psbt).unwrap();
-
-        let payjoin_proposal_psbt = payjoin.prepare_psbt(payjoin_proposal_psbt).unwrap();
-        debug!("Receiver's Payjoin proposal PSBT: {:#?}", payjoin_proposal_psbt);
-
-        base64::encode(&payjoin_proposal_psbt.serialize())
+        let payjoin_proposal = payjoin
+            .finalize_proposal(
+                |psbt: &Psbt| {
+                    receiver
+                        .wallet_process_psbt(
+                            &bitcoin::base64::encode(psbt.serialize()),
+                            None,
+                            None,
+                            Some(false),
+                        )
+                        .map(|res| res.psbt)
+                        .map_err(|e| Error::Server(e.into()))
+                },
+                Some(1),
+            )
+            .unwrap();
+        let psbt = payjoin_proposal.get_payjoin_psbt();
```

Comment:
Is there a good reason to return a PayjoinProposal to hang onto instead of just the psbt straight away? there might be.

Reply:
yea we might need to return other stuff in the future so In that case it wouldnt be a breaking change

Reply:
That makes sense. It might make more sense to name this `psbt()` since we know it's in the PayjoinProposal and `get_` is associated with returning from an index in rust by convention https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter